code/common/pubsub.q

```diff
@@ -73,13 +73,14 @@ selfiltered:{[x;y]
 
 // Publish table data
 pub:{[t;x]
-  if[not count x;:()];
-  if[count h:subrequestall[t];-25!(h;(`upd;t;x))];
-  if[t in subrequestfiltered`tbl;
-    {[t;x] data:eval(?;t;x`filts;0b;x`columns);neg[x`handle](`upd;t;data)}[t;] 
-      each select handle,filts,columns from subrequestfiltered where tbl=t
-  ];
- };
+    if[not count x;:()];
+    if[count h:subrequestall[t];-25!(h;(`upd;t;x))];
+    if[t in .stpps.subrequestfiltered`tbl;
+         {[t;x;sels] data:eval(?;x;sels`filts;0b;sels`columns);
```

Comment:
I know this is how it was before, but is there a reason to use `eval` here rather than just directly doing the functional select?

```suggestion
         {[t;x;sels] data:?[x;sels`filts;0b;sels`columns];
```

Reply:
Hi Jonathon, I tried changing the code to this suggestion and it was returning a `'type` error when I tried doing a force publish to check it was working, but it runs without issue using the `eval` line so I'm assuming that's why it is used and will just leave it the way it is